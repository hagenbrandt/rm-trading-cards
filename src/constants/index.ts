export const apiUri = 'https://rickandmortyapi.com/graphql';
