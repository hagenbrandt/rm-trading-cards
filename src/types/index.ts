export type CharacterJSON = {
  __typename: string;
  id: string;
  image: string;
  name: string;
};

export type HeadlineTypes = 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6';

export type ID = {
  id: string;
};

export type Image = {
  url: string;
  width: number;
  height: number;
};
