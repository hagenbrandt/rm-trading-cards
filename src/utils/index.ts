import { ApolloClient, InMemoryCache } from '@apollo/client';

/**
 * Creates an Apollo Client to fetch data. The passed
 * uri is used to fetch data from this adress.
 * @param {string} apiUri - the uri to fetch data from
 */
export const getApolloClient = (apiUri: string) => {
  if (!apiUri.includes('http')) {
    return;
  }

  return new ApolloClient({
    uri: apiUri,
    cache: new InMemoryCache(),
  });
};
