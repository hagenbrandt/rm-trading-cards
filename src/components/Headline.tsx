import { HeadlineTypes } from '../types';

export type HeadlineProps = {
  headlineText: string;
  headlineType: HeadlineTypes;
  headlineClass?: string;
};

const Headline = ({
  headlineText,
  headlineType,
  headlineClass,
}: HeadlineProps) => {
  if (!headlineText || !headlineType) {
    return <></>;
  }

  const headlineWithText = {
    h1: <h1 className={headlineClass ?? ''}>{headlineText}</h1>,
    h2: <h2 className={headlineClass ?? ''}>{headlineText}</h2>,
    h3: <h3 className={headlineClass ?? ''}>{headlineText}</h3>,
    h4: <h4 className={headlineClass ?? ''}>{headlineText}</h4>,
    h5: <h5 className={headlineClass ?? ''}>{headlineText}</h5>,
    h6: <h6 className={headlineClass ?? ''}>{headlineText}</h6>,
  };

  return headlineWithText[headlineType] || headlineWithText['h1'];
};

export default Headline;
