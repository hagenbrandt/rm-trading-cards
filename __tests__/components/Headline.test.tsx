import { render, screen, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';
import Headline from '../../src/components/Headline';
import { HeadlineTypes } from '../../src/types';

describe('Headline', () => {
  const headlineTypes: HeadlineTypes[] = ['h1', 'h2', 'h3', 'h4', 'h5', 'h6'];
  const headlineText = 'This is a headline';
  const headlineClassName = 'headline';
  const component = (
    <Headline
      headlineText={headlineText}
      headlineType='h1'
      headlineClass={headlineClassName}
    />
  );
  it('redners given headline tag with given text', () => {
    headlineTypes.forEach((item) => {
      render(<Headline headlineText={headlineText} headlineType={item} />);
      expect(screen.getByRole('heading').tagName).toBe(item.toUpperCase());
      expect(screen.getByRole('heading')).toHaveTextContent(headlineText);
      cleanup();
    });
  });
  it('returns empty DOM element when no text is given', () => {
    const { container } = render(
      <Headline headlineText='' headlineType='h1' />
    );
    expect(container).toBeEmptyDOMElement();
  });
  it('returns empty DOM element when no headline-type is given', () => {
    const { container } = render(
      <Headline headlineText='' headlineType={'' as HeadlineTypes} />
    );
    expect(container).toBeEmptyDOMElement();
  });
  it('has className when given', () => {
    render(component);
    expect(screen.getByRole('heading')).toHaveClass(headlineClassName);
  });
  it('has no className when not given', () => {
    render(<Headline headlineText={headlineText} headlineType={'h1'} />);
    expect(screen.getByRole('heading')).not.toHaveClass(headlineClassName);
  });
  it('matches snapshot', () => {
    const { container } = render(component);
    expect(container).toMatchSnapshot();
  });
});
