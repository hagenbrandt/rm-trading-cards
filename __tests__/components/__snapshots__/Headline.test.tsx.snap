// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Headline matches snapshot 1`] = `
<div>
  <h1
    class="headline"
  >
    This is a headline
  </h1>
</div>
`;
